### PR TITLE
cleanup (i guess?) & content - med cat and med cat app thoughts

### PR DESCRIPTION
**Pull request by arisaema12**
## About The Pull Request
was originally just pronoun-ifying, but got carried away and ended up adding loads of random status constraints to the med cat file and new thoughts to both med cat and med cat app... oops

## Why This Is Good For ClanGen
med cats can no longer generate thoughts about kits/leader/deputy when there aren't any in the clan. new thoughts add variety to clangen.

plus scribble seemed like she needed more prs to look through :D /j

## Proof of Testing
charismatic_cat_new_thought

## Changelog/Credits

- medicine cats will no longer generate thoughts about kits/leader/deputy when there aren't any in the clan
- added new medicine cat and medicine cat apprentice thoughts

### CODE REVIEW

File: resources/dicts/thoughts/alive/medicine_cat.json [hunk 1]
```diff
@@ -73,6 +118,18 @@
             "calm"
         ]
     },
+    {
+        "id": "calm_medicine_cat_apprentice",
+        "thoughts": [
+            "Calms down an apprentice who is panicking other their minor injury"
```
Comment (selkirks) [inline resources/dicts/thoughts/alive/medicine_cat.json]: shld be 'over their minor injury'

File: resources/dicts/thoughts/alive/medicine_cat.json [hunk 2]
```diff
@@ -232,6 +300,15 @@
             "nervous"
         ]
     },
+    {
+      "id": "playful_medicine_cat",
+        "thoughts": [
+            "Keeps {PRONOUN/m_c/poss} patients distracted with silly games while {PRONOUN/m_c/subject} treat them"
```
Comment (selkirks) [inline resources/dicts/thoughts/alive/medicine_cat.json]: treat shld be verb tagged

File: resources/dicts/thoughts/alive/medicine_cat.json (hunk 1 above)
Comment (scribblecrumb) [reply resources/dicts/thoughts/alive/medicine_cat.json]: "their" also needs to be pronoun tagged

File: resources/dicts/thoughts/alive/medicine_cat.json [hunk 3]
```diff
@@ -86,7 +143,8 @@
     {
         "id": "charismatic_medicine_cat",
         "thoughts": [
-            "Is doing {PRONOUN/m_c/poss} daily checkup on the elders"
+            "Is doing {PRONOUN/m_c/poss} daily checkup on the elders",
+	    "Easily persuades another Clan's medicine cat to gift c_n herbs"
```
Comment (scribblecrumb) [inline resources/dicts/thoughts/alive/medicine_cat.json]: We have moon events that mirror this and are also capable of actually changing the clan's herb supply... perhaps we could reword this a bit? Perhaps into "Is considering the best way to ask another Clan for herbs"?  or "Is coming up with just the right words to ask another Clan's medicine cat for extra herbs"

File: resources/dicts/thoughts/alive/medicine_cat_apprentice.json [hunk 4]
```diff
@@ -273,5 +273,32 @@
         "random_living_status": [
             "living"
         ]
+    },
+    {
+        "id": "meanie_medicine_cat_apprentice_to_nice_med",
+        "thoughts": [
+            "Thinks r_c is far too soft with {PRONOUN/r_c/poss} patients",
+	    "Wishes r_c wasn't so soft",
+	    "Has to promice r_c {PRONOUN/m_c/subject} won't gather deathberries again"
```
Comment (scribblecrumb) [inline resources/dicts/thoughts/alive/medicine_cat_apprentice.json]:
```suggestion
	    "Has to promise r_c {PRONOUN/m_c/subject} won't gather deathberries again"
```

**Review by scribblecrumb — CHANGES_REQUESTED** [review]

File: resources/dicts/thoughts/alive/medicine_cat.json (hunk 1 above)
Comment (arisaema12) [reply resources/dicts/thoughts/alive/medicine_cat.json]: ah I thought I should leave it ambiguous as to which apprentice the thought refers to, but i'll change it to be pronoun tagged

**Review by scribblecrumb — APPROVED** [review]

**Pull request merged**